lib/datasource.js

```diff
@@ -224,7 +224,10 @@ function tryModules(names, loader) {
     try {
       mod = loader(names[m]);
     } catch (e) {
-      /* ignore */
+      if (e.code !== 'MODULE_NOT_FOUND') {
+        throw e;
```

Comment:
Building off your comment about checking the string for the module name:

We could still make that check here and if it fails, return a more generic error.
```
// snip
} catch (e) {
  if (e.code === 'MODULE_NOT_FOUND') {
    var isMissing = _.some(names, function(name) {
      return _.contains(message, name);
    });
    if (!isMissing) {
      // If none of them match, then the message isn't what we expected.
      e.message = util.format("One or more dependencies could not be found.",
      "See stack trace for details.");
    }
  }
  throw e;
}
```

The downside is that this will run into false positives (modules with names that are common English words), and is a much more involved check.